hudi-common/src/main/java/org/apache/hudi/common/model/DefaultHoodieRecordPayload.java

```diff
@@ -68,17 +67,31 @@ public DefaultHoodieRecordPayload(Option<GenericRecord> record) {
     /*
      * We reached a point where the value is disk is older than the incoming record.
      */
-    eventTime = Option.ofNullable(getNestedFieldVal(incomingRecord, hoodieConfig
-        .getString(HoodiePayloadProps.PAYLOAD_EVENT_TIME_FIELD_PROP_KEY), true));
+    updateEventTime(incomingRecord, properties);
 
     /*
      * Now check if the incoming record is a delete record.
      */
-    if (isDeleteRecord(incomingRecord)) {
+    return isDeleteRecord(incomingRecord) ? Option.empty() : Option.of(incomingRecord);
+  }
+
+  @Override
+  public Option<IndexedRecord> getInsertValue(Schema schema, Properties properties) throws IOException {
+    if (recordBytes.length == 0) {
       return Option.empty();
-    } else {
-      return Option.of(incomingRecord);
     }
+    GenericRecord incomingRecord = incomingRecord(schema);
+    updateEventTime(incomingRecord, properties);
+
+    return isDeleteRecord(incomingRecord) ? Option.empty() : Option.of(incomingRecord);
+  }
+
+  private GenericRecord incomingRecord(Schema schema) throws IOException {
+    return bytesToAvro(recordBytes, schema);
+  }
```

Comment:
feel like this method is redundant. it only invokes `bytesToAvro`, which is more obvious to reader what happened. the variable name `incomingRecord` already tells what is returned.